project/backend/database/models.py

```diff
@@ -57,26 +58,41 @@ class Admin(BasicUser):
     def __str__(self):
         return self.user.first_name + " " + self.user.last_name
     
+class EnumRequest(enum.Enum):
+    WAITING  = 1
+    APPROVED = 2
+    DENIED   = 3  
+    
+    @classmethod
+    def choices(cls):
+        return [(key.value, key.name) for key in cls]
+    
 class Request(models.Model):
     """
      This class definition is written beforehand (to be implemented afterwards) 
      in order to be referred from other classes. e.g. ReviewRequest
+     
     """
-    pass
+    status = models.IntegerField(choices=EnumRequest.choices(), default=EnumRequest.WAITING.value)
+    def approve(self):
+        self.status = EnumRequest.APPROVED.value
+
+    def reject(self):
+        self.status = EnumRequest.DENIED.value
+
 class ReviewRequest(Request):
     """
      This class definition is written beforehand (to be implemented afterwards) 
      in order to be referred from other classes. e.g. Reviewer, Contributor
     """
-
-    # Note that reviewer is accessed directly by Reviewer instance,
-    # not via "receiverUserID" as proposed in project class diagram.
-    reviewer = models.ForeignKey(Reviewer, on_delete=models.CASCADE)
-    pass
-
-
-
-
+    reviewer  = models.ForeignKey(Reviewer, on_delete=models.CASCADE)     #Note that reviewer  is accessed directly by Reviewer  instance not via "receiverUserID" as proposed in project class diagram.
```

Comment:
this can be achieved by a method with a @property decoraiton (this method can fetch the receiver of the parent request model and return the reviewer subobject of it). this line causes duplicate data storage in the database with this implementation.